pkg/controller/registry/resolver/step_resolver.go

```diff
@@ -37,23 +35,30 @@ type OperatorStepResolver struct {
 	subLister              v1alpha1listers.SubscriptionLister
 	csvLister              v1alpha1listers.ClusterServiceVersionLister
 	client                 versioned.Interface
-	kubeclient             kubernetes.Interface
 	globalCatalogNamespace string
 	satResolver            *SatResolver
 	log                    logrus.FieldLogger
 }
 
 var _ StepResolver = &OperatorStepResolver{}
 
-func NewOperatorStepResolver(lister operatorlister.OperatorLister, client versioned.Interface, kubeclient kubernetes.Interface,
-	globalCatalogNamespace string, provider RegistryClientProvider, log logrus.FieldLogger) *OperatorStepResolver {
+func NewOperatorStepResolver(lister operatorlister.OperatorLister, client versioned.Interface, globalCatalogNamespace string, provider RegistryClientProvider, log logrus.FieldLogger) *OperatorStepResolver {
+	cacheSourceProvider := &mergedSourceProvider{
+		sps: []cache.SourceProvider{
+			SourceProviderFromRegistryClientProvider(provider, log),
+			&csvSourceProvider{
```

Comment:
Do we need to make an exported constructor for this so we can instantiate a CSVSourceProvider outside of this resolution package?

Reply:
No, the CSV would also leave this package when that happens.